irods/manager/user_manager.py

```diff
@@ -134,6 +134,10 @@ def modify_password(self, old_value, new_value, modify_irods_authentication_file
         """
         with self.sess.pool.get_connection() as conn:
 
+            if old_value != self.sess.pool.account.password or not isinstance(new_value, str) \
+                                                            or not(3 <= len(new_value) <= MAX_PASSWORD_LENGTH - 8):
+                raise Bad_password_change_parameter
```

Comment:
The integer codes was a last minute idea. Basically, the dev catches one exception and interrogates it for the exact failure reason.

Introducing more exceptions works just as well though.

Reply:
On the other hand, in this particular case, it may be better to let the client guess what they did wrong.  Give the hacker as few clues as possible.

In the name of slightly increased security and all ... : )

Reply:
I think I agree.

Please mark as resolved when ready.

Reply:
It can always be improved upon later, enums and messages added and so forth, if we see a necessity.

Reply:
I agree.